docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,82 @@
+|                                    |                   |                         |
+| try-acquire referred-to            | unchanged         | increment, iff non-zero |
+|   obj via weak-ref                 |                   |                         |
+|                                    |                   |                         |
+| release referred-to obj ref        | unchanged         | decrement               |
+|   (decrements ref count of         |                   |                         |
+|   referred-to obj)                 |                   |                         |
+|                                    |                   |                         |
+| weak-ref acquire ref               | increment         | unchanged               |
+|                                    |                   |                         |
+| weak-ref release ref               | decrement         | unchanged               |
+|                                    |                   |                         |
+| Referred-to obj destruction        | decrement         | decrement               |
+|                                    |                   |                         |
+| Weak-ref obj destruction           | decrement         | unchanged               |
+|                                    |                   |                         |
+```
+
+**Weak-ref management API:**
+
+The weak-ref implementation extends the base "ebpf object" management layer and is thus available for all primary object types.  The weak-ref management API are prototyped in the ```epbf_object.h``` header and provides the following calls:
+
+
+```ebpf_weak_reference_t* ebpf_object_weak_reference_get_reference(ebpf_core_object_t* object);```
+<br>
+The caller typically begins by calling this API to get a weak-ref pointer to the primary object they are interested in. This call increments the ref count for the weak-ref but does not modify the primary object's ref-count.  The caller now has a 'token' that they can now use to (conditionally) get the primary object pointer, either right away or at some later point in time.
+<br><br>
+
+```ebpf_core_object_t* ebpf_object_weak_reference_get_object_reference(ebpf_weak_reference_t* weak_reference);```
+<br>
+The caller uses this call to try and get the primary object pointer.  The 'try' is deliberate in that this call is not guaranteed to return a valid pointer to the referred-to object as the referred-to object might have been destroyed (or is in the process of being destroyed).  In such cases, this call will return a NULL.  If the object is still around, this call increments the ref-count of the referred-to object and returns a valid object pointer to the caller. The ref-count of the weak-ref object is not modified in this entire process. In case of a valid pointer return, the caller is responsible for releasing the reference on the returned object.
```

Comment:
```suggestion
The caller uses this call to try to get the primary object pointer.  The 'try' is deliberate in that this call is not guaranteed to return a valid pointer to the referred-to object as the referred-to object might have been destroyed (or be in the process of being destroyed).  In such cases, this call will return a NULL.  If the object is still around, this call increments the ref-count of the referred-to object and returns a valid object pointer to the caller. The ref-count of the weak-ref object is not modified in this entire process. In case of a valid pointer return, the caller is responsible for releasing the reference on the returned object.
```

Reply:
fixed